### ISSUE TITLE
Port name missing considered an error

### ISSUE THREAD

**Issue by whites11** [derailed/popeye#40]
Hi, just a brainstorming here.

Why do you consider an unnamed port in a deployment/pod as an error?
Do you have a source on the documentation that supports this decision?

Thanks

**Comment by derailed** [comment]
@whites11 - Unnamed ports do not yield an error but a warning. There is nothing wrong with not naming ports and there are no K8s mandate to do so. In my experience, naming ports makes Kubernetes manifests clearer to the reader about the intent of a given port. To boot, it affords a decoupling of a container port number with the managing resource such as services which yields to more resilient deployments should a container exposed port changes. Hence I feel this is a good best practice and thus the warning. Does this make sense?

**Comment by whites11** [comment]
Yes, it does. But then I must be missing something.
This is the output I get from a deployment with an unnamed port:

```
DEPLOYMENTS (52 SCANNED)                                                      💥 0 😱 48 🔊 0 ✅ 4 7٪
┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅┅
  · bug-report/bug-report..........................................................................😱
    🐳 node
      😱 No resources defined.
      😱 Unnamed port `3000.
```

I thought 😱 was an error, not a warning. Am I wrong?

**Comment by derailed** [comment]
@whites11 - [Icons](https://github.com/derailed/popeye#report-morphology)

**Comment by whites11** [comment]
Thanks, I missed that.
By the way, I never see the explosion for some reason. Must be the font.

**Comment by derailed** [comment]
@whites11 I am not sure I understand what you mean? The section summary on your report above shows the error icon, so the icon is available. Are you saying the sanitizer is not reporting errors correctly?

**Comment by whites11** [comment]
Let me clarify. In my terminal emulator (konsole on arch Linux) the icon appears as an empty rectangle. It must be because the default font does not support that character.
Nevermind, it is off topic.

Back to this issue, what does everyone think about not counting the warnings when calculating the percentage? In my case my pods are 0% ok, but 95% of the reasons are the unnamed ports.

**Comment by derailed** [comment]
@whites11 I think this relates to issue #43. If we add a sanitizer classification, you could then tune the reports by configuration a severity level. In this case unnamed ports could be set to info level and hence don't count against your overall cluster score. Would this make sense?

**Comment by whites11** [comment]
@derailed totally, that would make sense to me

**Comment by derailed** [comment]
@whites11 Resolved in v0.4.1. Please see v0.4.0 release notes and caution on how to override the defaults in the spinach.yml file. Thank you!
